core/src/main/java/org/apache/druid/common/config/JacksonConfigManager.java

```diff
@@ -90,14 +100,15 @@ public JacksonConfigManager(
    *
    * @param key of the config to set
    * @param oldValue old config value. If not null, then the update will only succeed if the insert
-   *                 happens when current database entry is the same as this value. If null, then the insert
-   *                 will not consider the current database entry.
+   *                 happens when current database entry is the same as this value. Note that the current database
+   *                 entry (in array of bytes) have to be exactly the same as the array of bytes of this value for
+   *                 update to succeed. If null, then the insert will not consider the current database entry.
```

Comment:
nit: it might be worth calling out why it uses bytes to be resilient across serde instead of the old object

Reply:
Done